hack/tools/deployment/deployment-update.go

```diff
@@ -0,0 +1,295 @@
+	jiraTicket = flag.String("jira-ticket", "", "Uses the jira ticket to create a branch with the same name with the environment added on.")
+)
+
+func main() {
+	// Verify required flags are used
+	flag.Parse()
+	if ptr.Deref(file, "") != "" ||
+		ptr.Deref(env, "") != "" ||
+		ptr.Deref(oldTag, "") != "" ||
+		ptr.Deref(newTag, "") != "" ||
+		ptr.Deref(newCommit, "") != "" {
+		log.Fatalf("flags - file, env, old-tag, new-tag, new-commit - are required")
+	}
+
+	// Verify the right env value was used
+	if !isValidEnvironment(*env) {
+		log.Fatalf("invalid environment: %s", *env)
+	}
+	log.Print("Updating deployment spec for environment: ", *env)
+
+	if ptr.Deref(jiraTicket, "") != "" {
+		autoCommitChanges = true
+
+		err := gitCheckout(filepath.Dir(*file), "master")
+		if err != nil {
+			log.Fatalf("failed to checkout the master branch: %v", err)
+		}
+
+		branchName = *jiraTicket + "-" + *env
+
+		log.Printf("Creating git branch: %s", branchName)
+		err = gitCreateBranch(filepath.Dir(*file), branchName)
+		if err != nil {
+			log.Fatalf("failed to create branch: %v", err)
+		}
+	}
+
+	// Read the deployment spec
+	deploymentFileContents, err := os.ReadFile(*file)
+	if err != nil {
+		log.Fatal("failed to read deployment file: ", err)
+	}
+
+	// Split the deployment spec content into a slice of strings
+	lines := strings.Split(string(deploymentFileContents), "\n")
+
+	// Update the deployment spec based on the environment
+	log.Printf("Updating tag from %s to %s and updating SHA to %s", *oldTag, *newTag, *newCommit)
+	err = updateDeploymentByEnvironment(*env, lines, *oldTag, *newTag, *newCommit)
+	if err != nil {
+		log.Fatalf("failed to update deployment: %v", err)
+	}
+
+	// Join the deployment spec back into one file
+	output := strings.Join(lines, "\n")
+
+	// Write the modified contents back to the file
+	err = os.WriteFile(*file, []byte(output), 0644)
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+	}
+
+	if autoCommitChanges {
+		// Add the changes in git
+		log.Printf("Adding changes in git")
+		err = gitAdd(filepath.Dir(*file))
+		if err != nil {
+			log.Fatalf("Error adding changes in git: %v", err)
+		}
+
+		// Commit the changes in git
+		log.Printf("Committing changes in git")
+		err = gitCommit(filepath.Dir(*file), fmt.Sprintf("Bump HyperShift in %s to %s / %s", *env, *newTag, *newCommit))
+		if err != nil {
+			log.Fatalf("Error committing changes in git: %v", err)
+		}
+
+		// Push the changes up
+		log.Printf("Pushing changes in git")
+		err = gitPush(filepath.Dir(*file), "origin", branchName)
+		if err != nil {
+			log.Fatalf("Error pushing changes in git: %v", err)
+		}
+
+		// TODO automate the MR request
+		log.Printf("-------------Info for MR-------------")
+		log.Printf("MR Title: %s: Bump HyperShift in %s to %s / %s", *jiraTicket, *env, *newTag, *newCommit)
+
+		// Output the changes between tags for the merge request
+		cmd := exec.Command("go", "run", "../release/notes.go", "--from="+*oldTag, "--to="+*newTag)
```

Comment:
It can be super wonky to interrupt `go run` - can we get the release note code to be importable? If not easy, this is fine.

Reply:
Discussed privately in Slack. Really just looking for the quickest path to making this work. We will also have to refactor this file when the new way of rolling out HO is implemented so prefer to go with what we have now.